File: app/src/main/java/com/packg/easynotes/Elements/Element.java
package com.packg.easynotes.Elements;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import org.jetbrains.annotations.NotNull;

@Entity (tableName = "element_table")
public class Element {
    @PrimaryKey(autoGenerate = true)
    @NotNull
    protected int id;
    @ColumnInfo(name = "element_type")
    protected String type;
    @ColumnInfo(name = "parent_id")
    protected int parentId;

    public String getType(){
        return this.type;
    }
    public int getId(){
        return this.id;
    }
}

File: app/src/main/java/com/packg/easynotes/Elements/Element.kt
package com.packg.easynotes.Elements

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.*

@Entity
abstract class Element {
    /**
     * Indicates when the [Element] was created or when it was edited.
     */
    @ColumnInfo(name = "create_date") var createDate: Calendar = Calendar.getInstance()
    @ColumnInfo(name = "edited_date") var editedDate: Calendar = Calendar.getInstance()
}